src/Purestone/Random.hs

```diff
@@ -0,0 +1,26 @@
+-------------------------------------------------------------------------------
+-- Perpetual Haskelling Initiative                                           --
+-------------------------------------------------------------------------------
+-- This source code is licensed under the MIT licence found in the           --
+-- LICENSE file in the root directory of this source tree.                   --
+-------------------------------------------------------------------------------
+
+module Purestone.Random ( shuffleIO ) where 
+
+-------------------------------------------------------------------------------
+
+import System.Random.Shuffle (shuffle')
+import System.Random (newStdGen, getStdGen)
+
+-------------------------------------------------------------------------------
+
+-- | `shuffleIO` takes a list of values and shuffles them using `shuffle'`
+-- and the global random number generator
+shuffleIO :: [a] -> IO [a]
+shuffleIO xs = do
+    g <- getStdGen
+    -- `shuffle'` doesn't update the generator so we update it with `newStdGen`
+    newStdGen
+    return $ shuffle' xs (length xs) g
```

Comment:
There is already a `shuffleM` function in the [`random-shuffle`](https://hackage.haskell.org/package/random-shuffle-0.0.4/docs/System-Random-Shuffle.html) package that does this, you should just be able to delete this and use `shuffleM` where you used `shuffleIO`

Reply:
Yeah, since Handler doesn't implement MonadRandom I thought I couldn't use shuffleM but of course IO does implement MonadRandom